### PR TITLE
Add action and first structure

### PR DESCRIPTION
**Pull request by patrickelectric**

### CODE REVIEW

File: .github/workflows/action.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        toolchain: stable
+        override: true
+    - name: Build docs
+      run: cargo doc
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v3
+      if: ${{ github.ref == 'refs/heads/master' }}
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./target/doc
```
Comment (joaoantoniocardoso) [inline .github/workflows/action.yml:37]: image

File: .github/workflows/action.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+    - uses: actions/checkout@master
+    - uses: actions-rs/toolchain@v1.0.6
+      with:
+        toolchain: stable
+        override: true
+    - name: Check Type
+      run: cargo fmt -- --check
+    - name: Run internal tests
+      run: cargo test --verbose -- --nocapture
```
Comment (joaoantoniocardoso) [inline .github/workflows/action.yml:17]: Use Clippy :)

```suggestion
      run: cargo test --verbose -- --nocapture
    - name: Check clippy
      uses: actions-rs/clippy-check@v1
      with:
        token: ${{ secrets.GITHUB_TOKEN }}
        args: --all-features
```

**Review by joaoantoniocardoso — APPROVED** [review]
Cool, just some optional points

**Review by RaulTrombin — APPROVED** [review]

File: .github/workflows/action.yml (hunk 2 above)
Comment (patrickelectric) [reply .github/workflows/action.yml:21]: Added

**Pull request merged**